extensions/cache/deployment/src/main/java/io/quarkus/cache/deployment/CacheProcessor.java

```diff
@@ -74,9 +75,9 @@ ValidationErrorBuildItem validateBeanDeployment(ValidationPhaseBuildItem validat
     }
 
     @BuildStep
-    @Record(STATIC_INIT)
+    @Record(RUNTIME_INIT)
     void recordCachesBuild(CombinedIndexBuildItem combinedIndex, BeanContainerBuildItem beanContainer, CacheConfig config,
```

Comment:
If we initialize the caches at runtime, it opens the gate for making the cache configuration partly runtime? At least the size of the caches?

Reply:
Yes, it does. I would probably make the entire [CaffeineConfig](https://github.com/quarkusio/quarkus/blob/master/extensions/cache/deployment/src/main/java/io/quarkus/cache/deployment/CacheConfig.java#L29) class runtime config. WDYT?

Reply:
exactly what I had in mind. But let's make it another PR once this one is in.